version.go

```diff
@@ -198,7 +198,16 @@ func compare(v1, v2 string) int {
 		strings2 = append([]string{""}, strings2...)
 	}
 
-	for i := 0; ; i++ {
+	var index int
+	c1 := len(strings1)
+	c2 := len(strings2)
+	if c1 >= c2 {
+		index = c1
+	} else {
+		index = c2
+	}
```

Comment:
What if comparing numbers1 with numbers2, and strings1 with strings2?

```suggestion
	if reflect.DeepEqual(numbers1, numbers2) && reflect.DeepEqual(strings1, strings2) {
		return 0
	}
```